lib/happo/uploader.rb

```diff
@@ -18,6 +19,9 @@ def upload_diffs
 
       bucket = find_or_build_bucket
       dir = SecureRandom.uuid
+      if !(@s3_bucket_path.nil? || @s3_bucket_path.empty?)
+        dir = File.join(@s3_bucket_path, SecureRandom.uuid)
```

Comment:
Using an `else` would also allow you to make the condition positive, which is easier to grok.

```rb
dir = if @s3_bucket_path.nil? || @s3_bucket_path.empty?
        SecureRandom.uuid
      else
        File.join(@s3_bucket_path, SecureRandom.uuid)
      end
```